tests/python/pants_test/base/context_utils.py

```diff
@@ -1,131 +1,7 @@
-
-    def report_target_info(self, scope, target, keys, val): pass
-
-
-  class TestLogger(logging.getLoggerClass()):
-    """A logger that converts our structured records into flat ones.
-
-    This is so we can use a regular logger in tests instead of our reporting machinery.
-    """
-
-    def makeRecord(self, name, lvl, fn, lno, msg, args, exc_info, *pos_args, **kwargs):
-      # Python 2 and Python 3 have different arguments for makeRecord().
-      # For cross-compatibility, we are unpacking arguments.
-      # See https://stackoverflow.com/questions/44329421/logging-makerecord-takes-8-positional-arguments-but-11-were-given.
-      msg = ''.join([msg] + [a[0] if isinstance(a, (list, tuple)) else a for a in args])
-      args = []
-      return super(TestContext.TestLogger, self).makeRecord(
-        name, lvl, fn, lno, msg, args, exc_info, *pos_args, **kwargs)
-
-  def __init__(self, *args, **kwargs):
-    super().__init__(*args, **kwargs)
-    logger_cls = logging.getLoggerClass()
-    try:
-      logging.setLoggerClass(self.TestLogger)
-      self._logger = logging.getLogger('test')
-    finally:
-      logging.setLoggerClass(logger_cls)
-
-  @contextmanager
-  def new_workunit(self, name, labels=None, cmd='', log_config=None):
-    """
-    :API: public
-    """
-    sys.stderr.write('\nStarting workunit {}\n'.format(name))
-    yield TestContext.DummyWorkUnit()
-
-  @property
-  def log(self):
-    """
-    :API: public
-    """
-    return self._logger
-
-  def submit_background_work_chain(self, work_chain, parent_workunit_name=None):
-    """
-    :API: public
-    """
-    # Just do the work synchronously, so we don't need a run tracker, background workers and so on.
-    for work in work_chain:
-      for args_tuple in work.args_tuples:
-        work.func(*args_tuple)
-
-  def subproc_map(self, f, items):
-    """
-    :API: public
-    """
-    # Just execute in-process.
-    return list(map(f, items))
-
-
-def create_context_from_options(options, target_roots=None, build_graph=None,
-                                build_configuration=None, address_mapper=None,
-                                console_outstream=None, workspace=None, scheduler=None):
-  """Creates a ``Context`` with the given options and no targets by default.
-
-  :param options: An :class:`pants.option.options.Option`-alike object that supports read methods.
-
-  Other params are as for ``Context``.
-  """
-  run_tracker = TestContext.DummyRunTracker()
-  target_roots = maybe_list(target_roots, Target) if target_roots else []
-  return TestContext(options=options, run_tracker=run_tracker, target_roots=target_roots,
-                     build_graph=build_graph, build_configuration=build_configuration,
-                     address_mapper=address_mapper, console_outstream=console_outstream,
-                     workspace=workspace, scheduler=scheduler)
+from pants.testutils.base.context_utils import TestContext as TestContext  # noqa
```

Comment:
What is the need for the apparently redundant aliasing? Afaict its not needed technically - a proof:
```
$ cat a/b/c/d.py 
from os import path
import sys


def func():
  return 42 
$ cat e/f/g.py 
from a.b.c.d import func, path, sys


print(f'{func()} {sys.executable} {path.basename(sys.executable)}')
$ python -c 'import e.f.g'
42 /usr/bin/python python
```

If it's for emphasis that these imports are just for re-exporting symbols in a new namespace a boilerplate comment on each file seems more savory (subjective), than a bunch of isort suppressions.

Reply:
Iirc, I believe we have a MyPy lint to require explicit exports in this style. 

The noqa is to suppress pycodestyle saying that we don’t use the import. We would need those no matter what. 

(On my phone cleaning old apartment - pardon me not confirming this all)

Reply:
Aha. OK, I'm happy to see this change, so not a huge deal, but a pretty explicit statement of all this then would be:
```python
from a import b
...
from foo.bar import Baz


b = b
...
Baz = Baz
```
That seems pretty clearly to be importing for the purpose of re-export in the current __name__ space with no violations of any linter's sensibilities.

I'll underscore though that I'm happy enough with all this.